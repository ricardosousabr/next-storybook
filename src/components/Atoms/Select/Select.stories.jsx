import Select from "./index.jsx";

export default {
  title: "Components/Select",
  Component: Select,
  parameters: {
    backgrounds: {
      values: [
        { name: "dark", value: "#000000" },
        { name: "white", value: "#ffffff" },
        { name: "purple", value: "#b220d7" },
      ],
    },
  },
};

export const Primary = (args) => <Select {...args} />;
Primary.args = {
  name: "",
  children: "",
  size: 0,
  autofocus: true,
  disabled: false,
  multiple: false,
};
