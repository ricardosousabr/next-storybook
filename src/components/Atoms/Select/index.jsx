import PropTypes from "prop-types";

export default function Select({children, name, autofocus, disabled, multiple, size }) {
  return (
    <select autofocus={autofocus} name={name} disabled={disabled} multiple={multiple} size={size} >
      <option>1213</option>
      <option>1213</option>
      <option>1213</option>
    </select>
  )
}

Select.propTypes ={
  name: PropTypes.string,
  children: PropTypes.string,
  size: PropTypes.number,
  autofocus: PropTypes.boolean,
  disabled: PropTypes.boolean,
  multiple: PropTypes.boolean,
}