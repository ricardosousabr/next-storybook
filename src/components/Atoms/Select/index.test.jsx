import renderer from "react-test-renderer";
import Select from "./index";

test("render text", () => {
  const tree = renderer.create(<Select />).toJSON();
  expect(tree).toMatchSnapshot();
});
