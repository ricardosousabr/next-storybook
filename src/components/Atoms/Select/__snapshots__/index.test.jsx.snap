// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render text 1`] = `
<select>
  <option>
    1213
  </option>
  <option>
    1213
  </option>
  <option>
    1213
  </option>
</select>
`;
